dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -315,60 +300,42 @@ public Object toMappedObj(Contentlet con) throws DotMappingException {
 	}
 
 	@SuppressWarnings("unchecked")
-	protected void loadCategories(Contentlet con, Map<String,String> m) throws DotDataException, DotSecurityException {
-		// first we check if there is a category field in the structure. We don't hit db if not needed
-		boolean thereiscategory=false;
-		Structure st=CacheLocator.getContentTypeCache().getStructureByInode(con.getStructureInode());
-		List<Field> fields=FieldsCache.getFieldsByStructureInode(con.getStructureInode());
-		for(Field f : fields)
-			thereiscategory = thereiscategory ||
-					f.getFieldType().equals(FieldType.CATEGORY.toString());
-
-		String categoriesString="";
-
-		if(thereiscategory) {
-			String categoriesSQL = "select category.category_velocity_var_name as cat_velocity_var "+
-					" from  category join tree on (tree.parent = category.inode) join contentlet c on (c.inode = tree.child) " +
-					" where c.inode = ?";
-			DotConnect db = new DotConnect();
-			db.setSQL(categoriesSQL);
-			db.addParam(con.getInode());
-			ArrayList<String> categories=new ArrayList<String>();
-			List<HashMap<String, String>> categoriesResults = db.loadResults();
-			for (HashMap<String, String> crow : categoriesResults)
-				categories.add(crow.get("cat_velocity_var"));
-
-			categoriesString=UtilMethods.join(categories, " ").trim();
-
-			for(Field f : fields) {
-				if(f.getFieldType().equals(FieldType.CATEGORY.toString())) {
-					String catString="";
-					if(!categories.isEmpty()) {
-						String catId=f.getValues();
-
-						// we get all subcategories (recursive)
-						Category category=APILocator.getCategoryAPI().find(catId, APILocator.getUserAPI().getSystemUser(), false);
-						List<Category> childrens=APILocator.getCategoryAPI().getAllChildren(
-								category, APILocator.getUserAPI().getSystemUser(), false);
-
-						// we look for categories that match childrens for the
-						// categoryId of the field
-						ArrayList<String> fieldCategories=new ArrayList<String>();
-						for(String catvelvarname : categories)
-							for(Category chCat : childrens)
-								if(chCat.getCategoryVelocityVarName().equals(catvelvarname))
-									fieldCategories.add(catvelvarname);
-
-						// after matching them we create the JSON field
-						if(!fieldCategories.isEmpty())
-							catString=UtilMethods.join(fieldCategories, " ").trim();
-					}
-					m.put(st.getVelocityVarName() + "." + f.getVelocityVarName(), catString);
-				}
-			}
+	protected void loadCategories(final Contentlet con, final Map<String,String> m)
+			throws DotDataException, DotSecurityException {
+	    // first we check if there is a category field in the structure. We don't hit db if not needed
+
+	    final ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(con.getContentTypeId());
+	    List<com.dotcms.contenttype.model.field.Field> catFields = type.fields().stream()
+				.filter(field -> field instanceof CategoryField).collect(CollectionsUtils.toImmutableList());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'field' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)